### PR TITLE
Change validator adapter refactoring

### PR DESCRIPTION
**Pull request by dimagorelik**

### CODE REVIEW

File: main/src/test/java/com/kenshoo/pl/entity/internal/validators/PrototypeFieldsCombinationValidationAdapterTest.java [hunk 1]
```diff
@@ -38,12 +37,11 @@
     @Mock
     private CurrentEntityState currentState;
 
-    @InjectMocks
     private PrototypeFieldsCombinationValidationAdapter<TestEntity> adapter;
 
     @Before
     public void setUp(){
-        adapter = spy(new PrototypeFieldsCombinationValidationAdapter<>(validator, ImmutableMap.<EntityFieldPrototype<?>, EntityField<TestEntity, ?>>of(TestDataFieldPrototype.FIELD_1, TestEntity.FIELD_1, TestDataFieldPrototype.FIELD_2, TestEntity.FIELD_2)));
+        adapter = new PrototypeFieldsCombinationValidationAdapter<>(validator, ImmutableMap.<EntityFieldPrototype<?>, EntityField<TestEntity, ?>>of(TestDataFieldPrototype.FIELD_1, TestEntity.FIELD_1, TestDataFieldPrototype.FIELD_2, TestEntity.FIELD_2));
```
Comment (effiban) [inline main/src/test/java/com/kenshoo/pl/entity/internal/validators/PrototypeFieldsCombinationValidationAdapterTest.java:44]: You didn't add a test here for the trigger as you did for the others - any reason?

Comment (dimagorelik) [reply main/src/test/java/com/kenshoo/pl/entity/internal/validators/PrototypeFieldsCombinationValidationAdapterTest.java:44]: @effiban 
I haven't added new tests. I changed the existing ones. I will add missing tests.

Comment (dimagorelik) [reply main/src/test/java/com/kenshoo/pl/entity/internal/validators/PrototypeFieldsCombinationValidationAdapterTest.java:45]: @effiban 
I added the missing test

**Comment by effiban** [comment]
@dimagorelik ine more thing - I see you didn't remove the old interface `EntityChangeValidator` for backwards-compatibility. That's fine but please mark it as deprecated with a comment linking to the new name.

**Comment by dimagorelik** [comment]
@effiban 
The interface EntityChangeValidator is not deprecated.  I only added the new interface for Adapter.

**Comment by effiban** [comment]
> @effiban
> The interface EntityChangeValidator is not deprecated. I only added the new interface for Adapter.

Who is still using it?

> @effiban
> The interface EntityChangeValidator is not deprecated. I only added the new interface for Adapter.

sorry, my mistake

**Review by effiban — APPROVED** [review]

File: main/src/main/java/com/kenshoo/pl/entity/internal/validators/ChangeValidatorAdapter.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * @return the supported change operation
+     */
+    SupportedChangeOperation getSupportedChangeOperation();
+
+    /**
+     * @return a list of fields to fetch
+     */
+    Stream<? extends EntityField<?, ?>> fetchFields();
```
Comment (galkoren) [inline main/src/main/java/com/kenshoo/pl/entity/internal/validators/ChangeValidatorAdapter.java:22]: fieldsToFetch !!

Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/internal/validators/ChangeValidatorAdapter.java]: @galkoren 
Done

**Pull request merged**